### PR TITLE
Refactor/condo/sberdoma 437/email normalization Added new unified mail normalization 

### PR DESCRIPTION
**Pull request by SavelevMatthew**
Added mail normalization, covered with tests, improved validations in Schemas.
For additional info visit problem on JIRA: https://doma.atlassian.net/browse/SBERDOMA-437

### CODE REVIEW

File: apps/condo/domains/common/utils/mail.spec.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    test('correct lowering data', () => {
+        expect(normalizeEmail('asd@asd.asd')).toEqual('asd@asd.asd')
+        expect(normalizeEmail('ASD@asd.asd')).toEqual('asd@asd.asd')
+        expect(normalizeEmail('asd@asD.aSd')).toEqual('asd@asd.asd')
+        expect(normalizeEmail('uBiIcAOnBoardINGoVVV2007@sber.doma')).toEqual('ubiicaonboardingovvv2007@sber.doma')
+    } )
+
+    test('multiple @  test', () => {
+        expect(normalizeEmail('asd@asd@ads')).toBeUndefined()
+        expect(normalizeEmail('asd@asd.asd')).toEqual('asd@asd.asd')
```
Comment (AntonAL) [inline apps/condo/domains/common/utils/mail.spec.js:24]: It seems, like this line is redundant, because another test case have it.

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/common/utils/mail.spec.js (hunk 1 above)
Comment (SavelevMatthew) [reply apps/condo/domains/common/utils/mail.spec.js]: removed :)

File: apps/condo/domains/common/utils/mail.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if (domainGroup.addresses.includes(domain)) return domainGroup.mainAddress
+    }
+    return domain
+}
+
+function processUserPart (userPart, domainPart) {
+    const beforePlus = userPart.split('+')[0]
+    if (domainPart !== 'gmail.com') return beforePlus
+    // For google abc.abc@ = abcabc@, but abc..abc != abcabc != abc.abc
+    return beforePlus.replace(/\.+/g, dotReplacer)
```
Comment (pahaz) [inline apps/condo/domains/common/utils/mail.js:50]:
```suggestion
    let result = userPart.split('+')[0]
    // For google abc.abc@ = abcabc@, but abc..abc != abcabc != abc.abc
    if (domainPart === 'gmail.com') result = result.replace(/\.+/g, dotReplacer)
    return result
```

It's better to add a link to the source if you take it from somewhere :)

File: apps/condo/domains/user/schema/User.js [hunk 3]
```diff
@@ -60,7 +62,12 @@ const User = new GQLListSchema('User', {
             hooks: {
                 resolveInput: ({ resolvedData }) => {
                     // TODO(pahaz): convert email to lover case is not enough! normalize by `+` and aliases (ya.ru/yandex.ru) and so on ..
```
Comment (pahaz) [inline apps/condo/domains/user/schema/User.js:64]: you can remove the todo

File: apps/condo/domains/common/utils/mail.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+const mailDomains = [
```
Comment (pahaz) [inline apps/condo/domains/common/utils/mail.js:1]: It's better to use UPPER_CASE for global consts

Comment (SavelevMatthew) [reply apps/condo/domains/common/utils/mail.js]: done

File: apps/condo/domains/user/schema/User.js (hunk 3 above)
Comment (SavelevMatthew) [reply apps/condo/domains/user/schema/User.js]: done

**Review by pahaz — APPROVED** [review]

**Pull request merged**